csrc/logical_domain_map.cpp

```diff
@@ -368,6 +368,52 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
     return dom_map;
   }
 
+  // TODO: refactor to use getNonMappingDomainInfo instead.
```

Comment:
Maybe it's time to switch to using the dispatch interface, either `OptIn` or `OptOut`.

Reply:
Interesting. Does that mean each operator would just have their custom rule for mapping, i.e. the special case that we are using here would become the right way for mapping then?